### DISCUSSION TITLE
Concepts shared between taskwarrior and timewarrior: description, tags, project

### DISCUSSION THREAD

**Discussion by mdouchin** [GothenburgBitFactory/timewarrior#432] [Q&A]
First, I would like to thank the devs for sharing this great set of tools. I am beginning to use task and timew, and would like to provide some feedback.

* **task**: I love the taskwarrior way of creating a new item: a task has got a **title** (description), some **tags** with `+`, some project with `pro:`. 
* **timew**: if I understand correctly, there is only the concept of **tags**

When using the taskwarrior hook to automatically add time entries when using start/stop in task, the task project, tags and title are `treated the same way` by timew, like tags. IMHO opinion, we loose some information here.

I would like to understand if this is a design choice, or if the same concepts will be added to timewarrior in the future ? This would help I think to create reports filters and totals: by projects, by tags, by project then tags, etc.

**Comment by lauft** [comment]
Thanks for your feedback and using Timewarrior! 🙂

You are right, Timewarrior currently only has the concept of **tags**, everything from Taskwarrior like projects is mapped to it. As far as I know this has mainly historical reasons (so there was some kind of design choice in the past).

Apart from this there are also some other issues around the connection between Taskwarrior and Timewarrior which require more details to be stored along Timewarrior intervals than just tags. Timewarrior's database format is currently not up to it, but there is an idea to migrate it to JSON (see #288). However this also requires a robust migration strategy for existing users.

So, yes, there are plans to add the ability to store additional metadata (apart from tags) along intervals, but it is currently not worked on.
